google/cloud/bigquery/magics/magics.py

```diff
@@ -596,6 +596,25 @@ def _cell_magic(line, query):
             _handle_error(error, args.destination_var)
             return
 
+        # Check if query is given as a reference to a variable.
+        if query.startswith("$"):
+            query_var_name = query[1:]
+
+            if query_var_name.isidentifier():
+                ip = IPython.get_ipython()
+                try:
+                    query = ip.user_ns[query_var_name]
+                except KeyError:
+                    raise NameError(
+                        f"Unknown query, variable {query_var_name} does not exist."
+                    ) from None
+                else:
+                    if not isinstance(query, (str, bytes)):
+                        raise TypeError(
+                            f"Query variable {query_var_name} must be string "
+                            "or bytes-like value."
```

Comment:
```suggestion
                            f"Query variable {query_var_name} must be a string "
                            "or a bytes-like value."
```

Reply:
Ah, and I already almost feel sorry for committing a suggestion, please appease the CLA bot. :)